### PR TITLE
Added Docker containers. Fixes 26

### PR DESCRIPTION
**Pull request by DavidHernandez**
- Upgraded ruby version to 2.3.1 as there is no official Docker Ruby container running 2.3.0
- Added -b 0.0.0.0 on the Rakefile to properly allow access to the container.
- Added docker-compose.yml to support Docker Compose and avoid usage of hard and difficult Docker commands to build and run the containers.
- Added .env.docker file with the Docker environment variables, so the users who dont want to use Docker, dont need to update the environment file.
- Added Dockerfile that installs all the dependencies. This is the only required file if you dont want to use Docker Compose.
- Added a start.sh script that updates the Gems only when the Gemfile changes, to avoid being required to rebuild the container. It also starts the ruby server on the container.

### CODE REVIEW

**Comment by DavidHernandez** [comment]
Seems that Travis is configured to run the tests over 2.3.0 and as I have changed the ruby version, it crashes before running the tests.

**Comment by hell03610** [comment]
update `.ruby-version` file ;)

File: Rakefile [hunk 1]
```diff
@@ -54,7 +54,7 @@ namespace :server do
       puts "\r\e[0KStopping ..."
       Rake::Task['server:down'].execute
     end
-    system "bundle exec rerun 'rackup -p $PORT -E $RACK_ENV'"
+    system "bundle exec rerun 'rackup -p $PORT -E $RACK_ENV -o 0.0.0.0'"
```
Comment (hell03610) [inline Rakefile:57]: is this fine for local environments?

Comment (DavidHernandez) [inline Rakefile:57]: I don't know. I haven't tried it on a local environment. I also don't have enough experience with ruby to know if there is a better way to do this without changing the Rakefile.

**Comment by carlosreig** [comment]
Yes, it works. I have just checked it.

This IP address is a constraint for incoming connections. So, if it is set to 127.0.0.1, then the server will only accept incoming connections from 127.0.0.1.

0.0.0.0 is like a wildcard to allow the incoming connections from anywhere.

File: Rakefile (hunk 1 above)
Comment (carlosreig) [inline Rakefile:57]: Yes, it works. I have just checked it.

This IP address is a constraint for incoming connections. So, if it is set to 127.0.0.1, then the server will only accept incoming connections from 127.0.0.1.

0.0.0.0 is like a wildcard to allow the incoming connections from anywhere.

**Pull request merged**